predict.py

```diff
@@ -166,12 +167,65 @@ def base_setup(
             shared_models=shared_models,
         )
 
+        if compile_fp8 or compile_bf16:
+            self.compile_ae()
+
         if compile_fp8:
             self.compile_fp8()
 
         if compile_bf16:
             self.compile_bf16()
 
+    @torch.inference_mode()
+    def compile_ae(self):
+        # helpful: export TORCH_COMPILE_DEBUG=1 TORCH_LOGS=dynamic,dynamo
+
+        # the order is important:
+        # torch.compile has to recompile if it makes invalid assumptions
+        # about the input sizes. Having higher input sizes first makes
+        # for fewer recompiles.
```

Comment:
any way we can compile once with craftier use of `dynamo.mark_dynamic` - add a `max=192` on dims 2 & 3? I assume you've tried this, curious how it breaks

Reply:
I tried `max=192`, but it didn't have any effect. Setting `torch.compile(dynamic=True)` makes for one fewer recompile, but I should check the runtime performance of that.